### ISSUE TITLE
Overflow of index in Redo feature

### ISSUE THREAD

**Issue by asl97** [r-40021/bingo#418]
When pressing the redo button after reaching the end, the `index` still increases instead of being capped at `e.props.bingoHistory.length-1`, resulting in requiring to press the undo button the same amount of time until the `index` is less than `e.props.bingoHistory.length` before undo has any effect

https://github.com/r-40021/bingo/blob/de3b026ca97fe878c27956959aa60a21d0247cc4/components/undo.js#L20-L29

![image](https://user-images.githubusercontent.com/4752645/213185053-6ba7c082-f957-450b-91f7-de843ca6c7a2.png)

```
Uncaught TypeError: Cannot read properties of undefined (reading 'max')  framework-114634acb84f8baa.js:formatted:1798
    at d (352-6c488188351b204e.js:formatted:590:41)
    at c (352-6c488188351b204e.js:formatted:580:17)
    at 65-265acea2814c5c71.js:308:5229
    at Array.some (<anonymous>)
    at 65-265acea2814c5c71.js:308:5211
    at Object.eV (framework-114634acb84f8baa.js:formatted:683:19)
    at eW (framework-114634acb84f8baa.js:formatted:701:16)
    at framework-114634acb84f8baa.js:formatted:1752:24
    at rn (framework-114634acb84f8baa.js:formatted:1763:14)
    at rt (framework-114634acb84f8baa.js:formatted:1792:29)
```

**Comment by asl97** [comment]
Oops, I shift+enter before writing the issue report, the issue report has been updated, sending this as a comment which I will delete afterwards to trigger a email notification with the full issue report.

When pressing the redo button after reaching the end, the `index` still increases instead of being capped at `e.props.bingoHistory.length-1`, resulting in requiring to press the undo button the same amount of time until the `index` is less than `e.props.bingoHistory.length` before undo has any effect

https://github.com/r-40021/bingo/blob/de3b026ca97fe878c27956959aa60a21d0247cc4/components/undo.js#L20-L29

![image](https://user-images.githubusercontent.com/4752645/213185053-6ba7c082-f957-450b-91f7-de843ca6c7a2.png)

```
Uncaught TypeError: Cannot read properties of undefined (reading 'max')  framework-114634acb84f8baa.js:formatted:1798
    at d (352-6c488188351b204e.js:formatted:590:41)
    at c (352-6c488188351b204e.js:formatted:580:17)
    at 65-265acea2814c5c71.js:308:5229
    at Array.some (<anonymous>)
    at 65-265acea2814c5c71.js:308:5211
    at Object.eV (framework-114634acb84f8baa.js:formatted:683:19)
    at eW (framework-114634acb84f8baa.js:formatted:701:16)
    at framework-114634acb84f8baa.js:formatted:1752:24
    at rn (framework-114634acb84f8baa.js:formatted:1763:14)
    at rt (framework-114634acb84f8baa.js:formatted:1792:29)
```

**Comment by r-40021** [comment]
This issue seems to be caused by a change in the Chakra UI specification.
It has been fixed in d61e658.
Thank you for your report!
